Add Cache SW manual test infrastructure

jridgewell opened the pull request:
Fixes #6495.

/cc @dvoytenko for the extensions script url changes

cramforce commented:
Looks good including the move of the extension code into a file with less deps. But don't you need to update some unit tests there?

jridgewell commented:
> But don't you need to update some unit tests there?

Yup, working on that next.

dvoytenko commented:
@jridgewell Are there actually any changes to how we calculate extension URL? Or is it just refactoring to a separate module?

jridgewell commented:
No changes to how they're done, just extracted and added [`calculateEntryPointScriptUrl`](https://github.com/ampproject/amphtml/pull/6532/files#diff-c5ad90666285b32b6075008f9ad89e58R65).

jridgewell commented:
PTAL

cramforce commented on test/functional/test-cache-sw-core.js:
```diff
@@ -108,7 +108,7 @@ runner.run('Cache SW', () => {
       });
 
-      it('rewrites v1 to versioned v1', () => {
+      it.skip('rewrites v1 to versioned v1', () => {
```
?

jridgewell replied:
Because we don't match `v1` right now, only `v0`.
